### ISSUE TITLE
Unable to cast object of type 'HashSet' to type 'IList'

### ISSUE THREAD

**Issue by peter-perot** [Tynamix/ObjectFiller.NET#68]
Hi,

I have an object with a property of type 'HashSet<string>'.

When I try to fill it, the following exception occurs:

Unable to cast object of type 'System.Collections.Generic.HashSet`1[System.String]' to type 'System.Collections.IList'.

In the source code a check is made against 'IEnumerable', and if true, a cast to 'IList' is made, which causes an exception.

**Comment by Tynamix** [comment]
Hey!
Thanks for your report, i will look for it!

**Comment by Tynamix** [comment]
Yes, thats a bug :disappointed: !
I will fix it with the next version. If you need it really fast, then you can use a custom RandomizerPlugin.
I wrote one for you:

```csharp
        public class HashSetPlugin<T> : IRandomizerPlugin<HashSet<T>>
        {
            private readonly int minCount;
            private readonly int maxCount;

            public HashSetPlugin(uint minCount = 0, uint maxCount = UInt32.MaxValue)
            {
                this.minCount = (int)minCount;
                this.maxCount = (int)maxCount;
            }

            public HashSet<T> GetValue()
            {
                int countOfItems = Randomizer<int>.Create(new IntRange(this.minCount, this.maxCount));
                var createdItems = Randomizer<T>.Create(countOfItems);

                HashSet<T> result = new HashSet<T>();

                foreach (var createdItem in createdItems)
                {
                    result.Add(createdItem);
                }

                return result;
            }
        }
```

You can then use it as below:
```csharp
        public class Bug68Poco
        {
            public HashSet<string> TestHashset { get; set; }
        }

        static void Main(string[] args)
        {

            Filler<Bug68Poco> filler = new Filler<Bug68Poco>();
            filler.Setup()
                   .OnType<HashSet<string>>().Use(new HashSetPlugin<string>(0, 100));

            Bug68Poco bug68Poco = filler.Create();

            foreach (string s in bug68Poco.TestHashset)
            {
                Console.WriteLine(s);
            }

            Console.ReadLine();
        }
```

Hope this helps right now. Will fix it in the next version!

**Comment by peter-perot** [comment]
Thank you, it works with the plugin. :-)

Q: In the setup chain you wrote 'OnType<HashSet<string>>'. Is there a way to configure it with kind of an open generic HashSet ('HashSet<>'), so that I'm not tied to string hash-sets?

**Comment by Tynamix** [comment]
hmm... i think this is not possible... But you are not really tied to string hashsets... You have just to setup the hashsets which have another typeparameter ... Like so:

```csharp
Filler<Bug68Poco> filler = new Filler<Bug68Poco>();
    filler.Setup()
            .OnType<HashSet<string>>().Use(new HashSetPlugin<string>(0, 100))
            .OnType<HashSet<int>>().Use(new HashSetPlugin<int>(0, 10))
            .OnType<HashSet<SomeObject>>().Use(new HashSetPlugin<SomeObject>());
```

Its a little bit of annoying, i know! But i will fix that in the next version! Then you will not need a seperate plugin for that!

**Comment by peter-perot** [comment]
TY. :+1:

**Comment by Tynamix** [comment]
will be fixed with #72

**Comment by Tynamix** [comment]
Fixed the bug. Will be released in next version!
